iterator.go

```diff
@@ -119,6 +119,10 @@ func (r RootLookuper) doExchange(ctx context.Context, m *dns.Msg,
 
 	resp, _, err := c.ExchangeContext(ctx, m, server)
 	if err := errors.ValidateResponse(server, resp, err); err != nil {
+		// We want the Authority section
+		if err.Err == "NODATA" {
```

Comment:
please use `errors.NODATA` instead of the literal string

Reply:
could this be cleaner?
```go
	e2 := errors.ValidateResponse(server, resp, err)
	switch {
	case e2 == nil:
		return resp, nil
	case e2.Err == errors.NODATA:
		// We want the Authority section
		return resp, nil
	default:
		// any other error
		return nil, e2
	}
```

Reply:
yes